### PR TITLE
Fix resetting of `LivelinessLostStatus::total_changes` [14984]

### PR DESCRIPTION
**Pull request by richiware**
## Description

RTPS layer always reset `LivelinessLostStatus::total_changes` after calling the `on_liveliness_lost()` listener callback.
This PR fixes this adding its own `LivelinessLostStatus` to `DataWriterImpl`.

@Mergifyio backport 2.5.x

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [ ] *n/a* Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [ ] *n/a* Any new/modified methods have been properly documented using Doxygen. 
- [ ] Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [ ] *n/a* Documentation builds and tests pass locally. 
- [ ] *n/a* New feature has been added to the `versions.md` file (if applicable).
- [ ] *n/a* New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: src/cpp/fastdds/publisher/DataWriterImpl.cpp [hunk 1]
```diff
@@ -1213,12 +1213,16 @@ void DataWriterImpl::InnerDataWriterListener::on_liveliness_lost(
...
     StatusMask notify_status = StatusMask::liveliness_lost();
     DataWriterListener* listener = data_writer_->get_listener_for(notify_status);
     if (listener != nullptr)
     {
-        listener->on_liveliness_lost(
-            data_writer_->user_datawriter_, status);
+        LivelinessLostStatus callback_status;
+        if (ReturnCode_t::RETCODE_OK == data_writer_->get_liveliness_lost_status(callback_status))
+        {
+            listener->on_liveliness_lost( data_writer_->user_datawriter_, status);
```
Comment (EduPonz) [inline src/cpp/fastdds/publisher/DataWriterImpl.cpp:1224]: Shouldn't we use the DW status?
```suggestion
            listener->on_liveliness_lost(data_writer_->user_datawriter_, callback_status);
```

File: src/cpp/fastdds/publisher/DataWriterImpl.cpp [hunk 2]
```diff
@@ -1566,6 +1568,14 @@ OfferedIncompatibleQosStatus& DataWriterImpl::update_offered_incompatible_qos(
     return offered_incompatible_qos_status_;
 }
 
+LivelinessLostStatus& DataWriterImpl::update_liveliness_lost_status(
+        const fastrtps::LivelinessLostStatus& liveliness_lost_status)
+{
+    liveliness_lost_status_.total_count = liveliness_lost_status.total_count;
+    liveliness_lost_status_.total_count_change += liveliness_lost_status.total_count;
```
Comment (EduPonz) [inline src/cpp/fastdds/publisher/DataWriterImpl.cpp:1575]: I think this is rather brittle as it very much relies on the implementation details on WLP, but I do not know what to do about that

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: src/cpp/fastdds/publisher/DataWriterImpl.cpp (hunk 2 above)
Comment (EduPonz) [inline src/cpp/fastdds/publisher/DataWriterImpl.cpp:1575]:
```suggestion
    liveliness_lost_status_.total_count_change += liveliness_lost_status.total_count_change;
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]

**Comment by MiguelCompany** [comment]
@mergifyio backport 2.6.x 2.3.x 2.1.x

**Pull request merged**